src/openrct2/world/Wall.cpp

```diff
@@ -35,26 +35,15 @@
  */
 void wall_remove_at(const CoordsXYRangedZ& wallPos)
 {
-    TileElement* tileElement;
-
-repeat:
-    tileElement = map_get_first_element_at(wallPos);
-    if (tileElement == nullptr)
-        return;
-    do
+    auto wallElement = map_get_wall_element_at(wallPos);
+    if (wallElement != nullptr)
     {
-        if (tileElement->GetType() != TILE_ELEMENT_TYPE_WALL)
-            continue;
-        if (wallPos.baseZ >= tileElement->GetClearanceZ())
-            continue;
-        if (wallPos.clearanceZ <= tileElement->GetBaseZ())
-            continue;
-
-        tile_element_remove_banner_entry(tileElement);
-        map_invalidate_tile_zoom1({ wallPos, tileElement->GetBaseZ(), tileElement->GetBaseZ() + 72 });
-        tile_element_remove(tileElement);
-        goto repeat;
-    } while (!(tileElement++)->IsLastForTile());
+        // Shoudn't we have a function to cast stuff back to TileElement*?
```

Comment:
TileElememt is the base class of WallElement you don't need to cast it will implicitly degrade to a TileElement pointer

Reply:
It seems that TileElementBase is the base class of WallElement not TileElement. I noticed that the classes are padded to all be the same size, so while rebasing the commit I substituted it for a simple reinterpret_cast.

Reply:
Argh yes I forgot its still got this awkward interface. Leave it with the reinterpret_casts it will be fixed in a later refactoring.